app/client/service_runner.py

```diff
@@ -88,7 +85,7 @@ def is_queue_empty():
             raise Exception('Master service did not become idle before timeout.')
 
     def kill(self):
-        self._run_service('{} stop'.format(self._main_executable))
+        self._run_service([self._main_executable, 'stop'])
 
     def _run_service(self, cmd, service_url=None):
```

Comment:
I see. It's just a wrapper around `subprocess.Popen` which has the exact same interface as Popen. And the only thing it does is to ensure all environment variables in the command will be expanded during execution time. https://docs.python.org/3.4/library/subprocess.html#popen-constructor